### ISSUE TITLE
Looped notification delivery

### ISSUE THREAD

**Issue by alxgsv** [ileitch/rapns#5]
Hello!

Thanks for simple to use gem. But i have such a problem:

When I create notification it start to deliver itstelf in loop. Typical loop in log is:

```
  SQL (0.1ms)  BEGIN
Mysql::Error: MySQL server has gone away: BEGIN
  Rapns::Notification Load (0.4ms)  SELECT `rapns_notifications`.* FROM `rapns_notifications` WHERE (deliver_after IS NULL)
  Rapns::Notification Load (0.4ms)  SELECT `rapns_notifications`.* FROM `rapns_notifications` WHERE (deliver_after < '2011-09-23 04:50:04')
  Rapns::Notification Load (0.3ms)  SELECT `rapns_notifications`.* FROM `rapns_notifications` WHERE `rapns_notifications`.`delivered` = 0 AND `rapns_notifications`.`failed` = 0
  Rapns::Notification Load (0.3ms)  SELECT `rapns_notifications`.* FROM `rapns_notifications` 
```

That is notification that was created (it not changes)

```
#<Rapns::Notification id: 6, badge: nil, device_token: "961acc448cf71ed21a6ca0d23835ed31e58a1428c2005d45ae7...", sound: "1.aiff", alert: "\xD0\x90\xD1\x82\xD1\x82\xD0\xB8\xD0\xBA\xD1\x83\xD1\x81 \xD0\xA4\xD0\xB8\xD0\xBD\xD1\x87 invited you to play Battles...", attributes_for_device: nil, expiry: 21600, delivered: false, delivered_at: nil, failed: false, failed_at: nil, error_code: nil, error_description: nil, deliver_after: nil, created_at: "2011-09-23 04:49:52", updated_at: "2011-09-23 04:49:52">
```

I am in development evnironment. What can be a problem?

**Comment by alxgsv** [comment]
More info — that problem occured on daemon that was launched yesterday at development mode.

When i restarted it — notifications start to deliver once. Should i restart daemon twice a day?

**Comment by ileitch** [comment]
Do you always get the error about the MySQL server having gone away? I'll take a look at this over the next few days, I think its a simple bug to fix - I just need to check when MySQL drops the connection and try to reconnect.

**Comment by alxgsv** [comment]
Yes, always in this loop (each loop iteration shows Mysql::Error: MySQL server has gone away: BEGIN)

**Comment by ileitch** [comment]
Fixed in v0.2.0
